cron/client.py

```
@@ -304,12 +317,16 @@ def do_measurement_control():
                 set_status('job_no')
-                if config['cluster']['client']['shutdown_on_job_no']:
-                    subprocess.check_output(['sync'], encoding='UTF-8', errors='replace')
-                    time.sleep(60) # sleep for 60 before going to suspend to allow logins to cluster when systems are fresh rebooted for maintenance
-                    subprocess.check_output(['sudo', 'systemctl', config['cluster']['client']['shutdown_on_job_no']], encoding='UTF-8', errors='replace')
 
                 if not args.testing:
+                    if reboot_after_seconds:= config['cluster']['client'].get('reboot_after_seconds'): # 0 will also resolve to false, which is what we want
+                        reboot_if_uptime_exceeded(reboot_after_seconds)
+
+                    if config['cluster']['client']['shutdown_on_job_no']:
+                        subprocess.check_output(['sync'], encoding='UTF-8', errors='replace')
```

Ruff (0.15.12)

[error] Starting a process with a partial executable path

(S607)